### ISSUE TITLE
fix: remove generic phospholipid reaction

### ISSUE THREAD

**Issue by BenjaSanchez** [SysBioChalmers/yeast-GEM#219]
### Description of the issue:

Reaction `r_4308` uses a generic `phospholipid` species, not specifying which type of phospholipid it is. As this creates confusion with the rest of the model, which has very specific details for each lipid, e.g. `1-phosphatidyl-1D-myo-inositol (1-16:1, 2-16:1)`, I would suggest removing that reaction until it can be formulated with more precise species. In any case, the reaction is blocked and both `phospholipid` species in it are dead-ends as they only appear in this reaction.

```
H2O [erm] + ATP [erm] + phospholipid [erm]  -> H+ [erm] + phosphate [erm] + phospholipid [gm] + ADP [erm]
```

@hongzhonglu what do you think?

**I hereby confirm that I have:**
- [x] Tested my code with [all requirements](https://github.com/SysBioChalmers/yeast-GEM#required-software---user) for running the model
- [x] Done this analysis in the `master` branch of the repository
- [x] Checked that a similar issue does not exist already

**Comment by hongzhonglu** [comment]
@BenjaSanchez Hi Ben,  I agree that r_4308 should be removed.

**Comment by edkerk** [comment]
Other instances of generic reactions:

### Generic lipid
rxnID | reaction name | equation
:--- |:--- |:---
r_4225 | Broad-range acid phosphatase   DET1 (EC 3.1.3.-) (Decreased ergosterol transport protein 1) | ATP[c] + H2O[c] + sterols[e]   <=> ADP[c] + H+[c] + phosphate[c] + sterols[c]
r_4251 | CTP:phosphatidate cytidyltransferase | CTP[m] + phosphatidate[m] <=> diphosphate[m] + H+[m] +   CDP-diacylglycerol[m]
r_4277 | acyl-CoA:sn-glycerol-3-phosphate 1-O-acyltransferase | Acyl-CoA[er] + 1-acyl-sn-glycerol 3-phosphate[er] <=>   coenzyme A[er] + 1,2-diacyl-sn-glycerol 3-phosphate[er]

---
### R-group
rxnID | reaction name | equation
:--- |:--- |:---
r_4199 | Glutathione S-transferase 1 (EC   2.5.1.18) (GST-I) | RX[er] + glutathione[er] <=>   HX[er] + R-S-glutathione[er]
r_4200 | Glutathione S-transferase 1 (EC 2.5.1.18) (GST-I) | glutathione[c] + RX[c] <=> HX[c] + R-S-glutathione[c]
r_4201 | Glutathione S-transferase 1 (EC 2.5.1.18) (GST-I) | glutathione[m] + RX[m] <=> HX[m] + R-S-glutathione[m]
r_4207 | Glutathione S-transferase omega-like 2 (EC 2.5.1.18)   (Extracellular mutant protein 4) (Glutathione-dependent dehydroascorbate   reductase) (EC 1.8.5.1) | RX[p] + glutathione[p] <=> HX[p] + R-S-glutathione[p]

---
### Generic protein
rxnID | reaction name | equation
:--- |:--- |:---
r_4233 | S-Adenosyl-L-methionine:protein-C-terminal-S-farnesyl-L-cysteine   O-methyltransferase | S-adenosyl-L-methionine[c] +   Protein C-terminal S-farnesyl-L-cysteine[c] <=>   S-adenosyl-L-homocysteine[c] + Protein C-terminal S-farnesyl-L-cysteine   methyl ester[c]
r_4239 | L-arginyl-tRNA(Arg):protein arginyltransferase | Arg-tRNA(Arg)[c] + Protein[c] <=> H+[c] + tRNA(Arg)[c] +   L-Arginyl-protein[c]
r_4240 | Dolichyl-diphosphooligosaccharide--protein glycosyltransferase   subunit STT3 (Oligosaccharyl transferase subunit STT3) (EC 2.4.99.18) | Protein asparagine[er] + G00008[er] <=> H+[er] +   Dolichyl diphosphate[er] + G00009[er]
r_4258 | [lipoyl-carrier protein]-L-lysine:lipoate ligase (AMP-forming) | ATP[c] + (R)-Lipoate[c] + Apoprotein[c] => AMP[c] +   diphosphate[c] + Protein N6-(lipoyl)lysine[c]
r_4280 | octanoyl-[acp]:protein N6-octanoyltransferase | octanoyl-ACP[m] + Apoprotein[m] <=> ACP1[m] + Protein   N6-(octanoyl)lysine[m]
r_4281 | lipoyl-[acp]:protein N6-lipoyltransferase | Apoprotein[m] + Lipoyl-[acp] [mitochondrion][m] <=>   ACP1[m] + Protein N6-(lipoyl)lysine[m]
r_4323 | protein N6-(octanoyl)lysine:sulfur sulfurtransferase | 2 S-adenosyl-L-methionine[m] + Protein N6-(octanoyl)lysine[m]   + 2 Sulfur donor[m] <=> 2 L-methionine[m] + Protein   N6-(lipoyl)lysine[m] + 2 5'-Deoxyadenosine[m]
r_4325 | Iron sulfur cluster assembly protein 2, mitochondrial (Iron   sulfur cluster scaffold protein 2) | S-sulfanyl-[L-cysteine desulfurase] [mitochondrion][m] +   [disordered-form [Fe-S] cluster scaffold protein] [mitochondrion][m] =>   S-sulfanyl-[cysteine desulfurase]-[disordered-form scaffold protein] complex   [mitochondrion][m]
r_4340 | D-amino-acid N-acetyltransferase HPA3 (DNT) (EC 2.3.1.36) (EC   2.3.1.48) (Histone and other protein acetyltransferase 3) | acetyl-CoA[c] + [protein]-L-lysine [cytoplasm][c] <=>   coenzyme A[c] + H+[c] + [protein]-N(6)-acetyl-L-lysine [cytoplasm][c]

---
### Generic sulfur
rxnID | reaction name | equation
:--- |:--- |:---
r_4173 | L-cysteine:sulfur-acceptor   sulfurtransferase | (sulfur carrier)-H[m] +   L-cysteine[m] <=> L-alanine[m] + (sulfur carrier)-SH[m]
-- | -- | --
r_4252 | Thiamine thiazole synthase (Thiazole biosynthetic enzyme) | L-glycine[c] + NAD[c] + Sulfur donor[c] <=> 3 H2O[c] +   nicotinamide[c] + ADP-5-ethyl-4-methylthiazole-2-carboxylate[c]
r_4324 | octanoyl-[acp]:sulfur sulfurtransferase | octanoyl-ACP[m] + 2 S-adenosyl-L-methionine[m] + 2 Sulfur   donor[m] <=> 2 L-methionine[m] + Lipoyl-[acp] [mitochondrion][m] + 2   5'-Deoxyadenosine[m]

---
### Various
rxnID | reaction name | equation
:--- |:--- |:---
r_4215 | Cys-Gly metallodipeptidase DUG1   (EC 3.4.13.-) (Deficient in utilization of glutathione protein 1) (GSH   degradosomal complex subunit DUG1) | H2O[c] + R-S-Cysteinylglycine[c]   <=> L-glycine[c] + S-Substituted L-cysteine[c]
r_4229 | Monocarboxylic acid amide amidohydrolase | H2O[c] + Monocarboxylic acid amide[c] <=> ammonium[c] +   Carboxylate[c]
r_4246 | Alpha-mannosidase (EC 3.2.1.24) (Alpha-D-mannoside   mannohydrolase) | H2O[v] + alpha-D-mannoside[v] => alpha-D-mannopyranose[v] +   non glycosylated sugar acceptor[v]

**Comment by hongzhonglu** [comment]
Some generic reactions can be changed into more detailed reactions if we know the detailed functions of the enzymes. So be careful when we delete some generic reactions.

**Comment by edkerk** [comment]
Good point @hongzhonglu, but I'd argue that protein-modifying enzymes are still outside the scope of a purely metabolic model, which is what yeast-GEM represents. So should we then remove the reactions labelled as `Generic protein` in the table above?

Regardless what we decide for the protein-modifying reactions, I think @hongzhonglu makes a fair argument that future knowledge might allow to curate these generic reactions to specific metabolites. In the meanwhile: would it be better to block these reactions (UB and LB = 0), to prevent any possibility of them being involved in some unrealistic loops? For instance `r_4252` and `r_4324` (Generic sulfer) are now strictly coupled as they are the only reactions using the `Sulfur donor[c]` metabolite.

- [ ] Keep `Generic protein` reactions?
- [ ] Block reactions with generic reactants?

**Comment by BenjaSanchez** [comment]
For now, blocking these reactions sounds like a good plan to me, maybe also mentioning in the corresponding notes the reason why?
